asset-properties-widget/component/asset-properties-view/asset-properties-view.service.ts

```diff
@@ -0,0 +1,159 @@
+import { Injectable } from '@angular/core';
+import {
+  AlarmService,
+  EventService,
+  IAlarm,
+  IEvent,
+  IManagedObject,
+  IMeasurement,
+  IOperation,
+  MeasurementService,
+  OperationService
+} from '@c8y/client';
+import {
+  AlarmRealtimeService,
+  DatePipe,
+  EventRealtimeService,
+  MeasurementRealtimeService,
+  OperationRealtimeService
+} from '@c8y/ngx-components';
+import { KPIDetails } from '@c8y/ngx-components/datapoint-selector';
+import { Observable, Subject } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+interface MeasurementValue {
+  unit?: string;
+  value: number;
+  date: string;
+  id: string;
+}
+@Injectable()
+export class AssetPropertiesViewService {
+  DEFAULT_FROM_DATE: string = '1970-01-01';
+  private variableSubject = new Subject<any>();
+  dateSet$ = this.variableSubject.asObservable();
+
+  constructor(
+    private c8yAlarms: AlarmService,
+    private c8yEvents: EventService,
+    private c8yMeasurements: MeasurementService,
+    private c8yOperation: OperationService,
+    private measurementRealtime: MeasurementRealtimeService,
+    private alarmRealtimeService: AlarmRealtimeService,
+    private eventRealtimeService: EventRealtimeService,
+    private operationRealtimeService: OperationRealtimeService,
+    private datePipe: DatePipe
+  ) {}
+
+  async getAlarms(filters): Promise<IManagedObject[]> {
+    let data = [];
+    await this.c8yAlarms.list(filters).then(function (alarms) {
+      data = alarms.data;
+    });
+    return data;
+  }
+
+  async getEvents(filters) {
+    let data = [];
+    await this.c8yEvents.list(filters).then(events => {
+      data = events.data;
+    });
+    return data;
+  }
+
+  async getMeasurements(filters) {
+    let data = [];
+    await this.c8yMeasurements.list(filters).then(measurements => {
+      data = measurements.data;
+    });
+    return data;
+  }
+
+  async getOperation(filters) {
+    let data = [];
+    await this.c8yOperation.list(filters).then(measurements => {
+      data = measurements.data;
+    });
+    return data;
+  }
+
+  async getLastDeviceMessage(device: IManagedObject) {
+    const filters = {
+      dateFrom: this.DEFAULT_FROM_DATE,
+      dateTo: this.datePipe.transform(
+        new Date().setDate(new Date().getDate() + 1),
+        'yyyy-MM-ddThh:mm:ssZZZZZ'
```

Comment:
use a constant with name that describes the dateformat.
Add a comment why the date is incremented by 1

Reply:
incorporated.